src/fireedge/src/modules/containers/VirtualMachines/VirtualMachines.js

```diff
@@ -0,0 +1,260 @@
+          props?.unselect && (propsInfo.unselect = props.unselect)
+          props.moreThanOneSelected &&
+            props.handleDismissError &&
+            (propsInfo.handleDismissError = props.handleDismissError)
+
+          return <InfoTabs {...propsInfo} />
+        }}
+      />
+    </TranslateProvider>
+  )
+}
+
+/**
+ * Displays details of a VM.
+ *
+ * @param {VM} vm - VM to display
+ * @param {Function} [gotoPage] - Function to navigate to a page of a VM
+ * @param {Function} [unselect] - Function to unselect a VM
+ * @returns {ReactElement} VM details
+ */
+const InfoTabs = memo(
+  ({ vm, gotoPage, unselect, handleDismissError, tags }) => {
+    const [getVm, { data: lazyData, isFetching }] = VmAPI.useLazyGetVmQuery()
+    const id = vm?.ID ?? lazyData?.ID
+    const RowActions = VmsTable.RowActions
+
+    const { settings: { FIREEDGE: fireedge = {} } = {} } = useAuth()
+    const { FULL_SCREEN_INFO } = fireedge
+    const { fullViewMode } = SERVER_CONFIG
+    const fullModeDefault =
+      FULL_SCREEN_INFO !== undefined
+        ? FULL_SCREEN_INFO === 'true'
+        : fullViewMode
+    const { isFullMode } = useGeneral()
+    const { setFullMode } = useGeneralApi()
+
+    useEffect(() => {
+      !isFullMode && gotoPage()
+    }, [])
+
+    return (
+      <Stack overflow="auto">
+        <Stack
+          direction="row"
+          alignItems="center"
+          justifyContent="space-between"
+          gap={1}
+          mx={1}
+          mb={1}
+        >
+          <Stack direction="row">
+            {fullModeDefault && (
+              <SubmitButton
+                data-cy="detail-back"
+                icon={<NavArrowLeft />}
+                tooltip={Tr(T.Back)}
+                isSubmitting={isFetching}
+                onClick={() => unselect()}
+              />
+            )}
+          </Stack>
+
+          <ButtonClearErrors
+            tags={tags}
+            handleDismissError={handleDismissError}
+          />
+
+          <Stack direction="row" alignItems="center" gap={1} mx={1} mb={1}>
+            {!fullModeDefault && (
+              <SubmitButton
+                data-cy="detail-full-mode"
+                icon={isFullMode ? <Collapse /> : <Expand />}
+                tooltip={Tr(T.FullScreen)}
+                isSubmitting={isFetching}
+                onClick={() => {
+                  setFullMode(!isFullMode)
+                }}
+              />
+            )}
+            {isFullMode && <RowActions vm={vm ?? lazyData} />}
+            <SubmitButton
+              data-cy="detail-refresh"
+              icon={<RefreshDouble />}
+              tooltip={Tr(T.Refresh)}
+              isSubmitting={isFetching}
+              onClick={() => getVm({ id })}
+            />
+            {typeof unselect === 'function' && (
+              <SubmitButton
+                data-cy="unselect"
+                icon={<Cancel />}
+                tooltip={Tr(T.Close)}
+                onClick={() => unselect()}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Using Function.prototype.bind and arrows functions as attributes will negatively impact performance in React. Each time the parent is rendered, the function will be re-created and trigger a render of the component causing excessive renders and more memory use.